test/account-add.test.js

```diff
@@ -206,4 +206,91 @@ test('account.add()', async (t) => {
 
     await p(carol.close)()
   })
+
+  await t.test(
+    "Can't publish with a key if the key has been del'd",
+    async (t) => {
+      rimraf.sync(DIR)
+
+      const keypair1 = Keypair.generate('ed25519', 'alice')
+      const keypair2 = Keypair.generate('ed25519', 'bob')
+
+      let peer = createPeer({ keypair: keypair1, path: DIR })
+
+      await peer.db.loaded()
+
+      const account = await p(peer.db.account.create)({
+        keypair: keypair1,
+        subdomain: 'person',
+      })
+      const accountMsg0 = peer.db.get(account)
+
+      const consent = peer.db.account.consent({ account, keypair: keypair2 })
+
+      const accountRec1 = await p(peer.db.account.add)({
+        account,
+        keypair: keypair2,
+        consent,
+        powers: ['external-encryption'],
+      })
+
+      const goodRec = await p(peer.db.feed.publish)({
+        account,
+        domain: 'post',
+        data: { text: 'potato' },
+        keypair: keypair2,
+      })
+      const postMootRec = peer.db.feed.findMoot(account, 'post')
+
+      const accountRoot = peer.db.get(account)
+      const delRec = await p(peer.db.feed.publish)({
+        account,
+        domain: accountRoot.metadata.domain,
+        keypair: keypair1,
+        data: {
+          action: 'del',
+          key: {
+            purpose: 'sig',
+            algorithm: 'ed25519',
+            bytes: keypair2.public,
+          },
+        },
+      })
+
+      const badRec = await p(peer.db.feed.publish)({
+        account,
+        domain: 'post',
+        data: { text: 'potato2' },
+        keypair: keypair2,
+      })
+
+      // Re-load as Carol, add the msgs to validate them
+      rimraf.sync(DIR)
+      const keypair3 = Keypair.generate('ed25519', 'carol')
+
+      const carol = createPeer({ keypair: keypair3, path: DIR })
+
+      await carol.db.loaded()
+
+      await p(carol.db.add)(accountMsg0, account)
+      await p(carol.db.add)(accountRec1.msg, account)
+      console.log(
+        'rec',
+        goodRec,
+        'rectangle',
+        goodRec.msg.metadata.tangles,
+        'moot',
+        postMootRec
+      )
+      await p(carol.db.add)(goodRec.msg, postMootRec.id)
```

Comment:
```
% node test/account-add.test.js
▶ account.add()
  ✔ Basic usage (414.540033ms)
  ✔ keypair with no "add" powers cannot add (793.603039ms)
  ✔ publish with a key in the account (531.707646ms)
rec {
  id: 'B1Qt4gRdfsjbvN6KeYJeVdDTXAk9sheFWKSKD4n8gZv4',
  msg: {
    data: { text: 'potato' },
    metadata: {
      dataHash: 'CvsZPpuqfYLEBC3TzbVMdxh91YEjhXQFe1Jdg46Zpn8L',
      dataSize: 17,
      account: 'jyUJX6tSFW3kU3vDSs79RRTqTtUuiW97upGLVRA61XF',
      accountTips: [Array],
      tangles: [Object],
      domain: 'post',
      v: 4
    },
    sigkey: 'FuUujJeb6TP7UAhWHi5HZMFP8fTejgPhnNyrx9c4GtwF',
    sig: '3UADDvQQ3QfcfC3joGzXQJeoako4V5e2TgugyfSk1CXGjroarrx2TWBXwLKWxtWiz3jMuuGVFgGeMES5iumyKF8o'
  },
  received: 1710714317072
} rectangle {
  '6fvVUU9WvbvUWWzggnHVMFsnWfbMutUBUXdCAR9jCYY6': {
    depth: 1,
    prev: [ '6fvVUU9WvbvUWWzggnHVMFsnWfbMutUBUXdCAR9jCYY6' ]
  }
} moot {
  id: '6fvVUU9WvbvUWWzggnHVMFsnWfbMutUBUXdCAR9jCYY6',
  msg: {
    data: null,
    metadata: {
      dataHash: null,
      dataSize: 0,
      account: 'jyUJX6tSFW3kU3vDSs79RRTqTtUuiW97upGLVRA61XF',
      accountTips: null,
      tangles: {},
      domain: 'post',
      v: 4
    },
    sigkey: 'FuUujJeb6TP7UAhWHi5HZMFP8fTejgPhnNyrx9c4GtwF',
    sig: '2vGBfd4L9xkzUJtMu9d2RL6h7YsM8XN3GsVZVv4x4juC8MUdLv5o5kgDh13AzVCsTKdhNFHf88nJPEQBRLmKmbP9'
  },
  received: 1710714317071
}
  ✖ Can't publish with a key if the key has been del'd (21.997003ms)
    Error: add() failed to verify msg
        at add (/home/me/prj/pzp/pzp-db/lib/index.js:570:17)
        at hooked (/home/me/prj/pzp/pzp-db/node_modules/hoox/index.js:10:15)
        ... 6 lines matching cause stack trace ...
        at async TestContext.<anonymous> (/home/me/prj/pzp/pzp-db/test/account-add.test.js:210:3)
        at async Test.run (node:internal/test_runner/test:632:9) {
      [cause]: Error: Invalid msg
          at verifyRec (/home/me/prj/pzp/pzp-db/lib/index.js:480:14)
          at add (/home/me/prj/pzp/pzp-db/lib/index.js:569:16)
          ... 5 lines matching cause stack trace ...
          at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
          at async Test.run (node:internal/test_runner/test:632:9)
          at async TestContext.<anonymous> (/home/me/prj/pzp/pzp-db/test/account-add.test.js:210:3) {
        [cause]: Error: Tangle "6fvVUU9WvbvUWWzggnHVMFsnWfbMutUBUXdCAR9jCYY6" is missing root message
            at get type [as type] (/home/me/prj/pzp/pzp-db/lib/msg-v4/tangle.js:229:13)
            at Object.validate (/home/me/prj/pzp/pzp-db/lib/msg-v4/validation.js:336:16)
            at verifyRec (/home/me/prj/pzp/pzp-db/lib/index.js:479:22)
            at add (/home/me/prj/pzp/pzp-db/lib/index.js:569:16)
            at hooked (/home/me/prj/pzp/pzp-db/node_modules/hoox/index.js:10:15)
            at node:internal/util:441:7
            at new Promise (<anonymous>)
            at hooked (node:internal/util:427:12)
            at TestContext.<anonymous> (/home/me/prj/pzp/pzp-db/test/account-add.test.js:285:28)
            at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      }
    }

▶ account.add() (1772.304172ms)
```

Reply:
@staltz am i doing something obviously wrong here?

Reply:
Hmm, `carol` has to `add()` the moot too.